crates/bevy_input/src/gamepad.rs

```diff
@@ -1648,130 +1778,161 @@ mod tests {
+            (0.84, Some(0.87777776)),  // scaled as: (0.84 - 0.05) / (0.95 - 0.05)
+            (0.43, Some(0.42222223)),  // scaled as: (0.43 - 0.05) / (0.95 - 0.05)
+            // barely above the low value means barely above 0 after scaling
+            (0.05001, Some(0.000011109644)), // scaled as: (0.05001 - 0.05) / (0.95 - 0.05)
+            // low enough to be rounded to 0 (dead zone)
             (0.05, Some(0.0)),
             (0.04, Some(0.0)),
             (0.01, Some(0.0)),
             (0.0, Some(0.0)),
+            // same exact tests as above, but below 0 (bottom half of the dead zone and live zone)
+            // low enough to be rounded to -1
             (-1.0, Some(-1.0)),
             (-0.99, Some(-1.0)),
             (-0.96, Some(-1.0)),
             (-0.95, Some(-1.0)),
-            (-0.9499, Some(-0.9499)),
-            (-0.84, Some(-0.84)),
-            (-0.43, Some(-0.43)),
-            (-0.05001, Some(-0.05001)),
+            // scaled inputs
+            (-0.9499, Some(-0.9998889)), // scaled as: (-0.9499 - -0.05) / (-0.95 - -0.05)
+            (-0.84, Some(-0.87777776)),  // scaled as: (-0.84 - -0.05) / (-0.95 - -0.05)
+            (-0.43, Some(-0.42222226)),  // scaled as: (-0.43 - -0.05) / (-0.95 - -0.05)
+            (-0.05001, Some(-0.000011146069)), // scaled as: (-0.05001 - -0.05) / (-0.95 - -0.05)
+            // high enough to be rounded to 0 (dead zone)
             (-0.05, Some(0.0)),
             (-0.04, Some(0.0)),
             (-0.01, Some(0.0)),
         ];
 
-        for (new_value, expected) in cases {
+        for (new_raw_value, expected) in cases {
             let settings = AxisSettings::new(-0.95, -0.05, 0.05, 0.95, 0.01).unwrap();
-            test_axis_settings_filter(settings, new_value, None, expected);
+            test_axis_settings_filter(settings, new_raw_value, None, expected);
         }
     }
 
     #[test]
-    fn test_axis_settings_default_filter_with_old_values() {
+    fn test_axis_settings_default_filter_with_old_raw_values() {
+        let threshold = 0.01;
+        // expected values are hardcoded to be rescaled to from 0.05..=0.95 to 0.0..=1.0
+        // new (raw), old (raw), expected
         let cases = [
-            (0.43, Some(0.44001), Some(0.43)),
-            (0.43, Some(0.44), None),
-            (0.43, Some(0.43), None),
-            (0.43, Some(0.41999), Some(0.43)),
-            (0.43, Some(0.17), Some(0.43)),
-            (0.43, Some(0.84), Some(0.43)),
-            (0.05, Some(0.055), Some(0.0)),
-            (0.95, Some(0.945), Some(1.0)),
-            (-0.43, Some(-0.44001), Some(-0.43)),
-            (-0.43, Some(-0.44), None),
-            (-0.43, Some(-0.43), None),
-            (-0.43, Some(-0.41999), Some(-0.43)),
-            (-0.43, Some(-0.17), Some(-0.43)),
-            (-0.43, Some(-0.84), Some(-0.43)),
-            (-0.05, Some(-0.055), Some(0.0)),
-            (-0.95, Some(-0.945), Some(-1.0)),
+            // enough increase to change
```

Comment:
I would suggest to split this single test into multiple small tests named like `enough_increase_to_change` instead of comments. But it's up to you since it was like this before.